lib/matplotlib/image.py

```diff
@@ -1291,24 +1291,34 @@ def imread(fname, format=None):
     """
     Read an image from a file into an array.
 
-    *fname* may be a string path, a valid URL, or a Python
-    file-like object.  If using a file object, it must be opened in binary
-    mode.
-
-    If *format* is provided, will try to read file of that type,
-    otherwise the format is deduced from the filename.  If nothing can
-    be deduced, PNG is tried.
-
-    Return value is a :class:`numpy.array`.  For grayscale images, the
-    return array is MxN.  For RGB images, the return value is MxNx3.
-    For RGBA images the return value is MxNx4.
-
-    matplotlib can only read PNGs natively, but if `PIL
-    <http://www.pythonware.com/products/pil/>`_ is installed, it will
-    use it to load the image and return an array (if possible) which
-    can be used with :func:`~matplotlib.pyplot.imshow`. Note, URL strings
-    may not be compatible with PIL. Check the PIL documentation for more
-    information.
+    Parameters
+    ----------
+    fname : str or file-like
+        The image file to read. This can be a filename, a URL or a Python
+        file-like object. If using a file-like object, it must be opened in
+        binary
+        mode.
+    format : str, optional
+        The image file format assumed for reading the data. If not
+        given, the format is deduced from the filename.  If nothing can
+        be deduced, PNG is tried.
+
+    Returns
+    -------
+    imagedata : :class:`numpy.array`
+        The image data. The returned array has shape
+
+        - (M, N) for grayscale images.
+        - (M, N, 3) for RGB images.
+        - (M, N, 4) for RGBA images.
+
+    Notes
+    -----
+    Matplotlib can only read PNGs natively. Further image formats are
+    supported via the optional dependency on `Pillow
+    <http://pillow.readthedocs.io/en/latest/>`_. Note, URL strings
+    may not be compatible with Pillow. Check the `Pillow documentation
+    <http://pillow.readthedocs.io/en/latest/>`_ for more information.
```

Comment:
I would not repeat the link, or, if you really want to do so, use and external hyperlink target (http://docutils.sourceforge.net/docs/user/rst/quickref.html#external-hyperlink-targets).  (In fact I would prefer *always* using external targets through the codebase as they make unrendered versions much more readable IMO, but that's a battle for another day.)